File: TheLionsDen.Model/Responses/RoomAmenityResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TheLionsDen.Model.Responses
{
    public class RoomAmenityResponse
    {
        public int RoomAmenitiesId { get; set; }
        public int AmenityId { get; set; }
        public int RoomId { get; set; }

        public string AmenityName =>Amenity.Name;

        public virtual AmenityResponse Amenity { get; set; }
    }
}

File: the_lions_den_mobile/lib/model/room_amenity/room_amenity_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:the_lions_den_mobile/model/amenity/amenity_response.dart';
import 'package:the_lions_den_mobile/model/room_type/room_type_response.dart';
part 'room_amenity_response.g.dart';

@JsonSerializable()
class RoomAmenityResponse {
  int? roomAmenityId;
  int? roomId;
  int? amenityId;
  String? amenityName;
  AmenityResponse? amenity;

  RoomAmenityResponse();

  factory RoomAmenityResponse.fromJson(Map<String, dynamic> json) =>
      _$RoomAmenityResponseFromJson(json);

  /// Connect the generated [_$RoomAmenityResponseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RoomAmenityResponseToJson(this);
}
